src/modules/twap/hooks/useFetchDiscreteOrders.ts

```diff
@@ -0,0 +1,81 @@
+import { useMemo } from 'react'
+
+import type { Order } from '@cowprotocol/contracts'
+import { OrderParameters, SupportedChainId } from '@cowprotocol/cow-sdk'
+
+import { useAsyncMemo } from 'use-async-memo'
+
+import { ComposableCoW } from 'abis/types'
+import { computeDiscreteOrderUid } from 'utils/orderUtils/computeDiscreteOrderUid'
+
+import { TradeableOrderWithSignature, useTwapOrdersTradeableMulticall } from './useTwapOrdersTradeableMulticall'
+
+import { TwapDiscreteOrderItem, TwapDiscreteOrders } from '../state/twapDiscreteOrdersAtom'
+import { TwapOrderInfo } from '../types'
+
+export function useFetchDiscreteOrders(
+  safeAddress: string,
+  chainId: SupportedChainId,
+  composableCowContract: ComposableCoW,
+  ordersInfo: TwapOrderInfo[]
+): TwapDiscreteOrders | null {
+  const ordersToVerifyParams = useMemo(() => {
+    return ordersInfo.map((info) => info.safeData.params)
+  }, [ordersInfo])
+
+  const ordersTradeableData = useTwapOrdersTradeableMulticall(safeAddress, composableCowContract, ordersToVerifyParams)
```

Comment:
Would this be re-executed for orders that we already did the multicall?
If so, shouldn't we do it only once?

Reply:
After the next iteration in twap order, the result of multicall will be different, it will return the next part.
I've added some optimisations:
Excluded `cancelled` and `expired` orders from multicall.